builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -795,6 +795,19 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 			networkInterfaces = append(networkInterfaces, networkInterface)
 		}
 	}
+	for _, v := range mvm.Guest.IpStack {
+		for _, route := range v.IpRouteConfig.IpRoute {
+			if route.Network == "0.0.0.0" {
+				deviceID, err := strconv.Atoi(route.Gateway.Device)
+				if err != nil {
+					log.Printf("[DEBUG] error at processing device id %#v: %#v", route.Gateway.Device, err)
+				} else {
+					log.Printf("[DEBUG] gateway of device id %d: %s", deviceID, route.Gateway.IpAddress)
```

Comment:
I think you're right about the first one. This should be a Info or something like that. Have to check how it's done at other places in TF.